geocoder_tester/base.py

```diff
@@ -20,6 +22,73 @@
 
 http = requests.Session()
 
+class GenericApi:
+    """ Access proxy for generic geocodejson APIs. The API URL must be
+        the search endpoint.
+
+        This class is here mainly for backward compatibility. You should
+        normally choose the specific API type you connect with.
+    """
+    def search_params(self, query, limit, lang, center):
+        params = {"q": query, "limit": limit}
+        if lang:
+            params['lang'] = lang
+        if center:
+            params['lat'] = center[0]
+            params['lon'] = center[1]
+
+        return params
+
+    def search_url(self):
+        return CONFIG['API_URL']
+
+
+class NominatimApi:
+    """ Access proxy for Nominatim APIs. The API URL must be the base
+        URL without /search or /reverse path.
+
+        Requires a Nominatim version which supports geocodejson.
+    """
+    def search_params(self, query, limit, lang, center):
+        params = {"format" : "geocodejson", "q" : query,
+                  "limit" : limit,
+                  "addressdetails" : "1"
+                 }
+        if lang:
+            params["accept-language"] = lang
+        # Nominatim has a bbox parameter which we could use here. However
+        # it is unclear how wide the bbox should extend.
+        if center:
+            skip(msg="API has no lat/lon search parameters")
+
+        return params
+
+    def search_url(self):
+        return CONFIG['API_URL'] + '/search'
+
+
+class PhotonApi:
+    """ Access proxy for Photon APIs. The API URL must be the base URL without
+        the /api or /reverse path.
+    """
+    def search_params(self, query, limit, lang, center):
```

Comment:
Humm, not sure why this method is extended here?

Reply:
Right, forgot the inheritance. Fixed as well.